src/assign_inliner.cpp

```diff
@@ -114,6 +128,22 @@ bool AssignInliner::can_inline(std::string key) {
           dynamic_cast<NumericLiteral*>(it->second.get()));
 }
 
+std::unique_ptr<Index> AssignInliner::visit(std::unique_ptr<Index> node) {
+  if (std::holds_alternative<std::unique_ptr<Identifier>>(node->value)) {
+    std::string key =
+        std::get<std::unique_ptr<Identifier>>(node->value)->toString();
+    if (this->can_inline(key)) {
+      std::unique_ptr<Expression> value = this->visit(assign_map[key]->clone());
+      if (auto ptr = dynamic_cast<Identifier*>(value.get())) {
+        value.release();
+        node->value = std::unique_ptr<Identifier>(ptr);
+      }
+    }
+    return node;
+  }
+  return Transformer::visit(std::move(node));
+}
+
```

Comment:
We need to do this here because Index doesn't have an Expression as it's value, but Slice has Expression for the type of value, so the visitor below works (ideally we should update Slice to be variant since we can't support general expressions for slices, but I'll leave that for a separate change set.  once we do that, we'll need a similar implementation as this method.